File: test-nvidia.py
from openai import OpenAI
import os
import sys

_USE_COLOR = sys.stdout.isatty() and os.getenv("NO_COLOR") is None
_REASONING_COLOR = "\033[90m" if _USE_COLOR else ""
_RESET_COLOR = "\033[0m" if _USE_COLOR else ""

client = OpenAI(
  base_url = "https://integrate.api.nvidia.com/v1",
  api_key = "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"
)


completion = client.chat.completions.create(
  model="meta/llama-3.1-70b-instruct",
  messages=[{"role":"user","content":"hello!"}],
  temperature=1,
  top_p=1,
  max_tokens=16384,
  seed=42,
  
  stream=True
)

for chunk in completion:
  if not getattr(chunk, "choices", None):
    continue
  if len(chunk.choices) == 0 or getattr(chunk.choices[0], "delta", None) is None:
    continue
  delta = chunk.choices[0].delta
  if getattr(delta, "content", None) is not None:
    print(delta.content, end="", flush=True)

File: scratch/test-nvidia.js
const nvidiaKey = "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";

async function testNvidiaVision() {
  console.log("Testing NVIDIA Vision API...");
  const start = Date.now();
  
  // Create a tiny 1x1 pixel base64 PNG for testing the image payload
  const dummyBase64 = "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=";
  
  try {
    const res = await fetch("https://integrate.api.nvidia.com/v1/chat/completions", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        "Authorization": `Bearer ${nvidiaKey}`
      },
      body: JSON.stringify({
        model: "meta/llama-3.2-11b-vision-instruct",
        messages: [
          {
            role: "user",
            content: [
              { type: "text", text: "Describe this image in JSON format: {'description': '...'}" },
              { type: "image_url", image_url: { url: `data:image/png;base64,${dummyBase64}` } }
            ]
          }
        ],
        temperature: 0,
        max_tokens: 100
      })
    });

    if (!res.ok) {
      console.error("NVIDIA API Error:", res.status, await res.text());
      return;
    }

    const data = await res.json();
    console.log(`Success! Took ${Date.now() - start}ms`);
    console.log("Response:", data.choices[0].message.content);
  } catch (err) {
    console.error("Fetch failed:", err);
  }
}

testNvidiaVision();
